File: #1.data-structures/#2.Stack/src/c#/src/MyStackProject/modules/Stack.cs
using System;
using System.Collections.Generic;

public class Stack<T>
{
  private readonly List<T> items = new();

  public void Push(T item) 
  {
    items.Add(item);
  }

  public T? Pop()
  {
    if(items.Count == 0) {
      return default;
    }

    var item = items[^1];
    items.RemoveAt(items.Count - 1);
    return item;
  }

  public T? Peek() 
  {
    return items.Count > 0 ? items [^1] : default;
  }

  public bool isEmpty()
  {
    return items.Count == 0;
  }

  public int Size () {
    return items.Count;
  }

  public void Clear() {
    items.Clear();
  }

  public IReadOnlyList<T> GetItems() {
    return items.AsReadOnly();
  }

}

File: #1.data-structures/#2.Stack/src/swfit/Sources/modules/Stack.swift
class Stack<T> {
    private var items: [T] = []

    func push(_ item: T) {
        items.append(item)
    }

    func pop() -> T? {
        return items.popLast()
    }

    func peek() -> T? {
        return items.last
    }

    func isEmpty() -> Bool {
        return items.isEmpty
    }

    func size() -> Int {
        return items.count
    }

    func clear() {
        items.removeAll()
    }

    func getItems() -> [T] {
        return items
    }
}
